Content.Server/Instruments/InstrumentSystem.cs

```diff
@@ -88,21 +204,83 @@ private void OnBoundUIOpened(EntityUid uid, InstrumentComponent component, Bound
         Clean(uid, component);
     }
 
+    private void OnBoundUIRequestBands(EntityUid uid, InstrumentComponent component, InstrumentBandRequestBuiMessage args)
+    {
+        foreach (var request in _bandRequestQueue)
+        {
+            // Prevent spamming requests for the same entity.
+            if (request.Entity == args.Entity)
+                return;
+        }
+
+        _bandRequestQueue.Add(args);
+    }
+
+    public (EntityUid, string)[] GetBands(EntityUid uid)
+    {
+        var metadataQuery = EntityManager.GetEntityQuery<MetaDataComponent>();
+
+        if (Deleted(uid, metadataQuery))
+            return Array.Empty<(EntityUid, string)>();
+
+        var list = new ValueList<(EntityUid, string)>();
+        var instrumentQuery = EntityManager.GetEntityQuery<InstrumentComponent>();
+
+        if (!TryComp(uid, out InstrumentComponent? originInstrument)
+            || originInstrument.InstrumentPlayer?.AttachedEntity is not {} originPlayer)
+            return Array.Empty<(EntityUid, string)>();
+
+        // It's probably faster to get all possible active instruments than all entities in range
+        var activeEnumerator = EntityManager.EntityQueryEnumerator<ActiveInstrumentComponent>();
+        while (activeEnumerator.MoveNext(out var entity, out _))
+        {
+            if (entity == uid)
+                continue;
+
+            // Don't grab puppet instruments.
+            if (!instrumentQuery.TryGetComponent(entity, out var instrument) || instrument.Master != null)
+                continue;
+
+            // We want to use the instrument player's name.
+            if (instrument.InstrumentPlayer?.AttachedEntity is not {} playerUid)
+                continue;
+
+            // Maybe a bit expensive but oh well GetBands is queued and has a timer anyway.
+            // Make sure the instrument is visible, uses the Opaque collision group so this works across windows etc.
+            if (!_interactions.InRangeUnobstructed(uid, entity, MaxInstrumentBandRange,
+                    CollisionGroup.Opaque, e => e == playerUid || e == originPlayer))
+                continue;
+
+            if (!metadataQuery.TryGetComponent(playerUid, out var playerMetadata)
+                || !metadataQuery.TryGetComponent(entity, out var metadata))
+                continue;
+
+            list.Add((entity, $"{playerMetadata.EntityName} - {metadata.EntityName}"));
+        }
+
+        return list.ToArray();
+    }
+
     public void Clean(EntityUid uid, InstrumentComponent? instrument = null)
     {
         if (!Resolve(uid, ref instrument))
             return;
 
         if (instrument.Playing)
         {
+            // Reset puppet instruments too.
```

Comment:
This reads like a TODO but is actually just a description of what the next line does. Can it be removed?

Reply:
No, otherwise it's not clear why the system reset event is sent at all.